Zend/zend_builtin_functions.c

```diff
@@ -510,7 +510,6 @@ ZEND_FUNCTION(define)
 
 register_constant:
 	/* non persistent */
-	ZEND_CONSTANT_SET_FLAGS(&c, CONST_CS, PHP_USER_CONSTANT);
```

Comment:
This macro also sets the module number (`PHP_USER_CONSTANT`) which must not be changed. So please do not remove the call itself, only get rid of `CONST_CS`. I guess this is the reason of the segfaults

Reply:
Call reverted.